app/src/componentsV2/CodeEditor/components/EditorV2/components/VariablePopOver/index.tsx

```diff
@@ -222,46 +229,101 @@ const VariableInfo: React.FC<{
+
+  const toggleVisibility = useCallback((key: InfoFieldLabel) => {
+    setRevealedLabels((prev) => ({ ...prev, [key]: !prev[key] }));
+  }, []);
 
   return (
     <>
       <div className="variable-info-property-container">
-        <span>{getScopeIcon(source.scope)} </span>
-        <span className="variable-header-info-seperator"> </span>
-        <div className="variable-info-header-name"> {source.name}</div>
+        <span>{getScopeIcon(source.scope)}</span>
+        <span className="variable-header-info-separator" />
+        <div className="variable-info-header-name">{source.name}</div>
 
-        <RQButton
-          type="transparent"
-          size="small"
-          icon={<MdEdit style={{ fontSize: "14px", color: "var(--requestly-color-text-subtle)" }} />}
-          onClick={onEditClick}
-          className="edit-variable-btn"
-        >
-          Edit
-        </RQButton>
+        <RoleBasedComponent resource="api_client_environment" permission="update">
+          <RQButton
+            type="transparent"
+            size="small"
+            icon={<MdEdit className="edit-icon" />}
+            onClick={onEditClick}
+            className="edit-variable-btn"
+          >
+            Edit
+          </RQButton>
+        </RoleBasedComponent>
       </div>
 
       <div className="variable-info-content-container">
         <div className="variable-info-content">
-          {infoFields.map(({ label, value }) => (
-            <React.Fragment key={label}>
-              <div className="variable-info-title">{label}</div>
-              <div className="variable-info-value">{value}</div>
+          {infoFields.map((field) => (
+            <React.Fragment key={field.label}>
+              <div className="variable-info-title">{field.label}</div>
+
+              <div className={`variable-info-value ${field.isSecret ? "with-toggle" : ""}`}>
+                <span className="value-content">
+                  {field.isSecret ? (
+                    revealedLabels[field.label] ? (
+                      <span className="secret-revealed">{String(field.value)}</span>
+                    ) : (
+                      <span className="secret-masked">{"•".repeat(Math.min(String(field.value).length, 15))}</span>
+                    )
+                  ) : (
+                    <span>{String(field.value)}</span>
+                  )}
+                </span>
+
+                {field.isSecret && (
+                  <RoleBasedComponent resource="api_client_environment" permission="update">
+                    <div className="eye-toggle-button">
+                      <RQButton
+                        type="transparent"
+                        size="small"
+                        icon={revealedLabels[field.label] ? <RiEyeLine /> : <RiEyeOffLine />}
+                        onClick={() => toggleVisibility(field.label)}
+                      />
+                    </div>
+                  </RoleBasedComponent>
+                )}
```

Comment:
Extract this into a reusable component.
props
1. value
2. roleResources

I guess rest all states can be managed inside the component